pub-matic-htb.js

```diff
@@ -72,8 +72,116 @@ function PubMaticHtb(configs) {
+                    h: sizes[1]
+                }
+            } else {
+                console.log("PubMatic: Error in sizes array");
+            }
+            retArr.push(impObj);
+        });
+        return retArr;
+    }
 
-    /**
+	function _parseSlotParam (paramName, paramValue) {
+      if (!Utilities.isString(paramValue)) {
+        paramValue && console.log('PubMatic: Ignoring param key: ' + paramName + ', expects string-value, found ' + typeof paramValue);
+        return undefined;
+      }
+
+      switch (paramName) {
+        case 'pmzoneid':
+          return paramValue.split(',')
+                    .slice(0, 50)
+                    .map(function(id) {
+                      return id.trim()
+                    })
+                    .join();
+        case 'kadfloor':
+        case 'lat':
+        case 'lon':
+          return parseFloat(paramValue) || undefined;
+        case 'yob':
+          return parseInt(paramValue) || undefined;
+        default:
+          return paramValue;
+      }
+    }
+
+    function __populateSiteObject(publisherId) {
+        var retObj =
+        {
+            page: Browser.topWindow.href,
+            ref: Browser.topWindow.document.referrer,
+            publisher: {
+                id: publisherId, // mandatory
+                domain: Browser.topWindow.location.hostname
+            },
+            domain: Browser.topWindow.location.hostname
+        }
+        return retObj;
+    }
+
+    function __populateDeviceInfo(rp){
+        var dnt = (Browser.topWindow.navigator.doNotTrack == 'yes' ||
+                    Browser.topWindow.navigator.doNotTrack == '1' ||
+                    Browser.topWindow.navigator.msDoNotTrack == '1')
+                    ? 1 : 0;
+        return {
+            ua: Browser.getUserAgent(),
+            js: 1,
+            dnt: dnt,
+            h: Browser.getScreenHeight(),
+            w: Browser.getScreenWidth(),
+            language: Browser.getLanguage(),
+            geo: {
+                lat: _parseSlotParam('lat', rp.lat),
+                lon: _parseSlotParam('lon', rp.lon),
+            }
+        }
+    }
+
+    function __populateUserInfo(rp) {
+        return {
+            gender: rp.gender ? rp.gender.trim() : undefined,
+            geo: {
+                lat: _parseSlotParam('lat', rp.lat),
+                lon: _parseSlotParam('lon', rp.lon),
+            },
+            yob: _parseSlotParam('yob', rp.yob)
+        };
+    }
+
+    function __populateExtObject(rp) {
+        var ext = {};
+        ext.wrapper = {};
+        ext.wrapper.profile = rp.profile || undefined; // remove ? check if mandatory
+        ext.wrapper.version = rp.version || undefined; // remove ? check if mandatory
+        ext.wrapper.wiid = rp.wiid || undefined; //
+        //ext.wrapper.wv = Constants.REPO_AND_VERSION;
+        ext.wrapper.transactionId = rp.transactionId;
+        ext.wrapper.wp = 'pbjs' ;
+
+        return ext;
+    }
+	/**
```

Comment:
The following functions: 
1. __populateExtObject
2. __populateUserInfo
3.  __populateDeviceInfo
4. __populateSiteObject

reference properties on the `returnParcels` object that doesn't exist. As a result, these values are `undefined`. 

In order to reference the properties provided in the `config` (spec/support/mockPartnerConfig.json), please reference the `config` object in the constructor and save the properties you need in instance variables (or one consoildated object). You can then reference these variables in their respective functions later.

Reply:
made the required changes. the additional properties are now read into globalConfigs instead of populating them in the returnParcels.

Reply:
made the required changes